core/src/main/java/org/infinispan/configuration/cache/Configurations.java

```diff
@@ -66,4 +67,12 @@ public static boolean hasSharedCacheLoaderOrWriter(Configuration configuration)
       }
       return false;
    }
+
+   public static boolean isRecoveryEnabled(Configuration configuration) {
+      TransactionConfiguration txConfig = configuration.transaction();
+      return txConfig.transactionMode() == TransactionMode.TRANSACTIONAL &&  //tx cache
+            !txConfig.useSynchronization() && //xa enlistment
+            txConfig.transactionProtocol() != TransactionProtocol.TOTAL_ORDER && //non TO cache
+            txConfig.recovery().enabled(); //recovery enabled
```

Comment:
We already validate the first 3 conditions in `RecoveryConfigurationBuilder`, do we need another check?